test/extended/cluster/cl.go

```diff
@@ -0,0 +1,147 @@
+package cluster
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	oapi "github.com/openshift/origin/pkg/api"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+	testutil "github.com/openshift/origin/test/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+const deploymentRunTimeout = 5 * time.Minute
+
+var _ = g.Describe("clusterloader", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLI("cl", exutil.KubeConfigPath())
+		//deploymentFixture = exutil.FixturePath("testdata", "deployments", "test-deployment-test.yaml")
+	)
+
+	var c *kclientset.Clientset
+	g.BeforeEach(func() {
+		c = oc.AdminKubeClient()
+		ParseConfig(e2e.TestContext.Viper)
+	})
+
+	g.It("Should load the cluster", func() {
+
+		project := ConfigContext.ClusterLoader.Projects
+		tuningSets := ConfigContext.ClusterLoader.TuningSets
+		if len(project) < 1 {
+			e2e.Failf("invalid config file.\nFile: %v", project)
+		}
+
+		//var namespaces []*projectapi.Project
+		//totalPods := 0 // Keep track of how many pods for stepping
+		// TODO sjug: add concurrency
+		for _, p := range project {
+			// Find tuning if we have it
+			tuning := GetTuningSet(tuningSets, p.Tuning)
+			if tuning != nil {
+				e2e.Logf("Our tuning set is: %v", tuning)
+			}
+			for j := 0; j < p.Number; j++ {
+				// Create namespaces as defined in Cluster Loader config
+				nsName := fmt.Sprintf("%s%d", p.Basename, j)
+				err := oc.Run("new-project").Args(nsName).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				e2e.Logf("%d/%d : Created new namespace: %v", j+1, p.Number, nsName)
+
+				//namespaces = append(namespaces, ns)
+
+				// Create templates as defined
+				for _, template := range p.Templates {
+					var allArgs []string
+					e2e.Logf("We're loading file %v: ", template.File)
+					templateObj, err := testutil.GetTemplateFixture(template.File)
+					if err != nil {
+						e2e.Failf("Cant read template config file. Error: %v", err)
+					}
+					allArgs = append(allArgs, templateObj.Name)
+
+					if template.Parameters == (ParameterConfigType{}) {
+						e2e.Logf("Pod environment variables will not be modified.")
+					} else {
+						params := convertVariablesToString(template.Parameters)
+						allArgs = append(allArgs, params...)
+					}
+
+					config, err := oc.AdminClient().Templates(nsName).Create(templateObj)
+					e2e.Logf("Template %v created, config: %+v", templateObj.Name, config)
+
+					err = oc.SetNamespace(nsName).Run("new-app").Args(allArgs...).Execute()
+					o.Expect(err).NotTo(o.HaveOccurred())
+				}
+				// This is too familiar, create pods
+				for _, pod := range p.Pods {
```

Comment:
Mostly historical reasons; the original clusterloader has support for individual objects (routes, services, pv/pvc's...).  We've mostly moved to templates at this point for everything.  @sjug is this here because of the kubecon demo?